### PR TITLE
add pytables recipe

### PR DESCRIPTION
**Pull request by msarahan**

### CODE REVIEW

**Comment by pelson** [comment]
Windows failing with:

```
C:\Program Files (x86)\Common Files\Microsoft\Visual C++ for Python\9.0\VC\Bin\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:"C:\Miniconda\envs\_build\Library\lib" /LIBPATH:"C:\Miniconda\envs\_build\Library\lib" /LIBPATH:C:\Miniconda\envs\_build\libs /LIBPATH:C:\Miniconda\envs\_build\PCbuild /LIBPATH:C:\Miniconda\envs\_build\PC\VS9.0 hdf5.lib /EXPORT:initutilsextension build\temp.win32-2.7\Release\tables\utilsextension.obj build\temp.win32-2.7\Release\src/utils.obj build\temp.win32-2.7\Release\src/H5ARRAY.obj build\temp.win32-2.7\Release\src/H5ATTR.obj build\temp.win32-2.7\Release\c-blosc/hdf5/blosc_filter.obj build\temp.win32-2.7\Release\c-blosc/blosc\blosc.obj build\temp.win32-2.7\Release\c-blosc/blosc\blosclz.obj build\temp.win32-2.7\Release\c-blosc/blosc\shuffle.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\lz4-r119\lz4.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\lz4-r119\lz4hc.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\snappy-1.1.1\snappy-c.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\snappy-1.1.1\snappy-sinksource.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\snappy-1.1.1\snappy-stubs-internal.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\snappy-1.1.1\snappy.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\adler32.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\compress.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\crc32.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\deflate.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\gzclose.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\gzlib.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\gzread.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\gzwrite.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\infback.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\inffast.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\inflate.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\inftrees.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\trees.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\uncompr.obj build\temp.win32-2.7\Release\c-blosc/internal-complibs\zlib-1.2.8\zutil.obj /OUT:build\lib.win32-2.7\tables\utilsextension.pyd /IMPLIB:build\temp.win32-2.7\Release\tables\utilsextension.lib /MANIFESTFILE:build\temp.win32-2.7\Release\tables\utilsextension.pyd.manifest
   Creating library build\temp.win32-2.7\Release\tables\utilsextension.lib and object build\temp.win32-2.7\Release\tables\utilsextension.exp
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_NATIVE_DOUBLE_g referenced in function ___pyx_pf_6tables_14utilsextension_20_broken_hdf5_long_double
utils.obj : error LNK2001: unresolved external symbol _H5T_NATIVE_DOUBLE_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_NATIVE_LDOUBLE_g referenced in function ___pyx_pf_6tables_14utilsextension_20_broken_hdf5_long_double
utils.obj : error LNK2001: unresolved external symbol _H5T_NATIVE_LDOUBLE_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_C_S1_g referenced in function ___pyx_pf_6tables_14utilsextension_54atom_to_hdf5_type
H5ATTR.obj : error LNK2001: unresolved external symbol _H5T_C_S1_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_UNIX_D64BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_UNIX_D32BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_IEEE_F64BE_g referenced in function _initutilsextension
utils.obj : error LNK2001: unresolved external symbol _H5T_IEEE_F64BE_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_IEEE_F32BE_g referenced in function _initutilsextension
utils.obj : error LNK2001: unresolved external symbol _H5T_IEEE_F32BE_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U64BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U32BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U16BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U8BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I64BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I32BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I16BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I8BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_B8BE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_UNIX_D64LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_UNIX_D32LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_IEEE_F64LE_g referenced in function _initutilsextension
utils.obj : error LNK2001: unresolved external symbol _H5T_IEEE_F64LE_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_IEEE_F32LE_g referenced in function _initutilsextension
utils.obj : error LNK2001: unresolved external symbol _H5T_IEEE_F32LE_g
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U64LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U32LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U16LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_U8LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I64LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I32LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I16LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_I8LE_g referenced in function _initutilsextension
utilsextension.obj : error LNK2019: unresolved external symbol _H5T_STD_B8LE_g referenced in function _initutilsextension
utils.obj : error LNK2019: unresolved external symbol _H5T_NATIVE_FLOAT_g referenced in function _create_ieee_float16
H5ARRAY.obj : error LNK2019: unresolved external symbol _H5P_CLS_DATASET_CREATE_ID_g referenced in function _H5ARRAYmake
blosc_filter.obj : error LNK2019: unresolved external symbol _H5E_ERR_CLS_g referenced in function _blosc_set_local
blosc_filter.obj : error LNK2019: unresolved external symbol _H5E_PLINE_g referenced in function _blosc_set_local
blosc_filter.obj : error LNK2019: unresolved external symbol _H5E_CALLBACK_g referenced in function _blosc_set_local
blosc_filter.obj : error LNK2019: unresolved external symbol _H5E_CANTREGISTER_g referenced in function _register_blosc
build\lib.win32-2.7\tables\utilsextension.pyd : fatal error LNK1120: 35 unresolved externals
error: command 'C:\\Program Files (x86)\\Common Files\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\link.exe' failed with exit status 1120
 
C:\Miniconda\conda-bld\work>if 0 == 1 (
rd /s /q C:\Miniconda\envs\_build\Lib\site-packages\numpy  
 rd /s /q C:\Miniconda\envs\_build\Lib\site-packages\__pycache__  
 rd /s /q C:\Miniconda\envs\_build\Lib\site-packages\Cython 
) 
Command failed: C:\Windows\system32\cmd.exe /c call bld.bat
```

File: recipes/pytables/bld.bat [hunk 1]
```diff
@@ -0,0 +1,26 @@
+REM Would be nice to include blosc support, but building this with external blosc is currently broken.
+REM    See https://github.com/PyTables/PyTables/issues/451
+
+
+REM rd /s /q c-blosc
+REM git clone https://github.com/Blosc/c-blosc
+REM cd c-blosc
+REM git checkout v1.7.0
+REM git clone https://github.com/Blosc/hdf5-blosc
```
Comment (pelson) [inline recipes/pytables/bld.bat]: Is it worth just packaging these?

File: recipes/pytables/bld.bat [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+REM cd ..
+
+REM %PYTHON% setup.py install --blosc=%LIBRARY_PREFIX% --hdf5=%LIBRARY_PREFIX% --bzip2=%LIBRARY_PREFIX%
+
+
+REM Proceed without external blosc
+
+%PYTHON% setup.py install --hdf5=%LIBRARY_PREFIX% --bzip2=%LIBRARY_PREFIX%
+
+if %PY3K%==1 (
```
Comment (pelson) [inline recipes/pytables/bld.bat]: A comment that this is a conda-build bug (and a link) would be healthy I think here.

File: recipes/pytables/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+{% set version="3.2.2" %}
+
+package:
+  name: pytables
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/PyTables/PyTables.git
+  git_tag: v.{{ version }}
```
Comment (pelson) [inline recipes/pytables/meta.yaml]: Would you mind getting the ```tar.gz``` associated with this tag?

File: recipes/pytables/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    # - blosc                  # [win]
+    - numpy
+  run:
+    - python
+    - argparse                 # [py26]
+    - numpy
+    - numexpr
+    - hdf5
+    - zlib
+    - setuptools
```
Comment (pelson) [inline recipes/pytables/meta.yaml]: Is this really a runtime dep too? You seem to have mapped the entrypoints nicely, so just wondering if you can remove setuptools here.

File: recipes/pytables/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  commands:
+    - pt2to3 -h
+    - ptdump -h
+    - ptrepack -h
+    - pttree -h
+
+about:
+  home: http://www.pytables.org
+  license: BSD
+  license_file: LICENSE.txt
```
Comment (pelson) [inline recipes/pytables/meta.yaml]: I don't think this file is in the PR.

File: recipes/pytables/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  summary: brings together Python, HDF5 and NumPy to easily handle large amounts of data
+
+extra:
+  recipe-maintainers:
+    - msarahan
```
Comment (pelson) [inline recipes/pytables/meta.yaml]: Just wondering if you have any plans to widen the maintainers list once the PR is ready to go?

File: recipes/pytables/bld.bat (hunk 1 above)
Comment (msarahan) [inline recipes/pytables/bld.bat]: See the comment a few lines up.  Pytables can't currently utilize these.  Yes, it's worth packaging them, and I'll do so if someone beats me to it. However, we can't use them here.

File: recipes/pytables/bld.bat (hunk 2 above)
Comment (msarahan) [inline recipes/pytables/bld.bat]: I'm not actually sure that's a bug.  I have no idea whether we should be packaging pycache files, but we seem to have the opinion that we should package pyc files.  I have removed that section for now.

Comment (jakirkham) [inline recipes/pytables/bld.bat]: I'm not even sure that we should be compiling all pyc files. That seems to cause a variety of problems.

File: recipes/pytables/meta.yaml (hunk 6 above)
Comment (msarahan) [inline recipes/pytables/meta.yaml]: Sure - I have added a couple more, happy to add anyone else as well.   You want on?

File: recipes/pytables/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    - unittest2                # [py26]
+  commands:
+    - pt2to3 -h
+    - ptdump -h
+    - ptrepack -h
+    - pttree -h
+
+about:
+  home: http://www.pytables.org
+  license: BSD
```
Comment (jakirkham) [inline recipes/pytables/meta.yaml]: Which version?

File: recipes/pytables/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    - numexpr
+    - hdf5
+    - zlib
+    - setuptools
+    - bzip2                    # [win]
+    # building blosc is done internally.
+    # this is an old version, though.
+    # would be better to link against new version externally.
+    # See https://github.com/PyTables/PyTables/issues/451
+    # - blosc                  # [win]
```
Comment (jakirkham) [inline recipes/pytables/meta.yaml]: Do we have `blosc` somewhere?

File: recipes/pytables/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  git_tag: v.{{ version }}
+  patches:
+    # remove march=native, fix library paths
+    - setup.patch          # [win]
+    # fix a syntax error
+    - blosc_macro.patch
+    # teach it to look for dlls in Library\bin on win
+    - tables_init.patch    # [win]
+    # remove march=native for more widely compatible builds
+    - march.patch          # [unix]
```
Comment (jakirkham) [inline recipes/pytables/meta.yaml]: Are there issues or PRs upstream for these changes?

File: recipes/pytables/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+requirements:
+  build:
+    - python
+    - cython
+    - zlib                     # [win]
+    - lzo                      # [linux]
+    - bzip2
+    - numexpr
+    - hdf5
+    - setuptools
```
Comment (jakirkham) [inline recipes/pytables/meta.yaml]: Do we require `numpy`, as well? I am guessing we will need `x.x` too.

**Comment by jakirkham** [comment]
Also, please take a look at our [pinning guide]( https://github.com/conda-forge/staged-recipes/wiki/Pinned-dependencies ).

File: recipes/pytables/meta.yaml (hunk 8 above)
Comment (msarahan) [inline recipes/pytables/meta.yaml]: It's in defaults, I think.  We can't use it on this recipe, though.  Please see that github issue.

**Comment by msarahan** [comment]
CIs seem to be having trouble downloading the source for some reason.  I'm hoping this will sort itself out with some time.

**Comment by jakirkham** [comment]
Yes, see this issue ( https://github.com/conda-forge/staged-recipes/issues/508 ).

**Comment by jakirkham** [comment]
It is only with AppVeyor though. I think we need to have someone take a look who has access to the settings. Should message support as well.

**Comment by msarahan** [comment]
Both circleCI and Travis are failing here.  Whatever source code they are getting is incorrect - it fails to extract.

**Comment by jakirkham** [comment]
I looked at the logs of both Travis CI and CircleCI. Seems to be an issue with the patch not applying correctly in both cases. So, not an infrastructure issue.

**Comment by msarahan** [comment]
yep, please look a little more carefully.  The patch is failing to apply because the source is not extracted: https://travis-ci.org/conda-forge/staged-recipes/jobs/127108881#L416

**Comment by jakirkham** [comment]
Hmm...downloading it from GitHub works fine outside of CIs. Not really sure what is wrong.

GitHub and all CIs status (can't find one for AppVeyor) appear to be green, but maybe there is a bigger issue at play here that just hasn't been discovered.

In any event, I'm going to grab some dinner and will check back in an hour to see if stuff is still off. If you know anything more by then, feel free to add it to this issue ( https://github.com/conda-forge/staged-recipes/issues/508 ).

**Comment by jakirkham** [comment]
One last thought, maybe we are being rate limited here. See this issue ( https://github.com/conda-forge/staged-recipes/issues/509 ).

**Comment by jakirkham** [comment]
Yeah, rate limiting is over so I don't think that is the cause. Could you try using the source from PyPI?

**Comment by jakirkham** [comment]
Toggled open status to get AppVeyor back up. Also, we had download issues last time this was touched so it is probably worth dusting off all the CIs.

File: recipes/pytables/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+about:
+  home: http://www.pytables.org
+  license: BSD 3-clause
+  summary: brings together Python, HDF5 and NumPy to easily handle large amounts of data
+
+extra:
+  recipe-maintainers:
+    - avalentino
+    - FrancescAlted
+    - jakirkham
```
Comment (jakirkham) [inline recipes/pytables/meta.yaml]: If it's all the same, I think I will pass on this one. While I use `h5py` quite a bit, I don't really use `pytables` and I may already be stretching myself thin.

**Comment by msarahan** [comment]
superseded by #1134 - closing.

**Pull request closed without merging**